test/testutil/driver.c

```diff
@@ -268,3 +268,39 @@ int run_tests(const char *test_prog_name)
     return EXIT_SUCCESS;
 }
 
+/*
+ * Glue an array of strings together and return it as an allocated string.
+ * Optionally return the whole length of this string in |out_len|
+ * Store and cache 64 first strlen() results avoid duplicates calls.
```

Comment:
I'm not thrilled with this.  Yes 64 is more than we'll ever use, but 64K was always enough for a PC too.  And to optimize out a strlen call seems like the wrong trade-off.

Reply:
I think it was 640K... You know, 1MB addressable, but 384K are taken by BIOSes and screen buffer...

Reply:
But lyrics aside, I'm with Rich on this one, it's not performance-critical path, so no real point to make it more complicated than it has to be. Besides, implementation is broken, first loops in pairs have to have additional loop termination conditions. If strlen is considered that repelling it would custom strcpy that returns pointer to the end of string could be a way...